contracts/test/StorageContractTest.t.sol

```diff
@@ -40,4 +44,94 @@ contract StorageContractTest is Test {
         (,, reward) = storageContract.miningRewards(0, 1);
         assertEq(reward, storageContract.paymentIn(PREPAID_AMOUNT + STORAGE_COST * 2, 0, 1));
     }
+
+    function testRewardMiner() public {
+        address miner = vm.addr(2);
+        uint256 mineTs = 10000;
+        uint256 diff = 1;
+
+        vm.expectRevert("StorageContract: not enough balance");
+        storageContract.rewardMiner(0, miner, mineTs, 1);
+
+        vm.deal(address(storageContract), 1000);
+
+        (,, uint256 reward) = storageContract.miningRewards(0, mineTs);
+        storageContract.rewardMiner(0, miner, mineTs, diff);
+        (uint256 l, uint256 d, uint256 b) = storageContract.infos(0);
+        assertEq(l, mineTs);
+        assertEq(d, diff);
+        assertEq(b, 1);
+        assertEq(miner.balance, reward);
+    }
+
+    function testReentrancy() public noGasMetering {
+        uint256 prefund = 1000;
+        // Without reentrancy protection, the fund could be drained by 29 times re-entrances given current params.
+        vm.deal(address(storageContract), prefund);
+        storageContract.setKvEntryCount(1);
+        Attacker attacker = new Attacker(storageContract);
+        vm.prank(address(attacker));
+
+        uint256 _blockNum = 1;
+        uint256 _shardId = 0;
+        uint256 _nonce = 0;
+        bytes32[] memory _encodedSamples = new bytes32[](0);
+        uint256[] memory _masks = new uint256[](0);
+        bytes memory _randaoProof = "0x01";
+        bytes[] memory _inclusiveProofs = new bytes[](0);
+        bytes[] memory _decodeProof = new bytes[](0);
+        // currently this error is not reachable on github server
```

Comment:
why it is not reachable on github server?

Reply:
I guess it is because `noGasMetering` does not work properly, which possibly related to https://github.com/foundry-rs/foundry/issues/3971. However, the workaround in the comments does not work. 

This issue happens on another linux too, but my local MBP works well.